src/main.c

```diff
@@ -517,19 +517,37 @@ int main(int argc, char **argv)
 			}
 			if (strcmp(long_options[option_index].name,
 			           "cookie") == 0) {
-				cli_cfg.cookie = strdup(optarg);
+				char *cookie_with_prefix;
+
+				cookie_with_prefix = strdup_with_prefix(optarg, "SVPNCOOKIE=");
+				if (cookie_with_prefix == NULL) {
+					log_error("Could not prepend \"SVPNCOOKIE=\" to the cookie.\n");
+					goto exit;
+				}
+				free(cli_cfg.cookie);
+				cli_cfg.cookie = cookie_with_prefix;
 				break;
 			}
 			if (strcmp(long_options[option_index].name,
 			           "cookie-on-stdin") == 0) {
-				char *cookie = read_from_stdin(COOKIE_SIZE);
+				char *cookie;
+				char *cookie_with_prefix;
 
+				cookie = read_from_stdin(COOKIE_SIZE);
 				if (cookie == NULL) {
-					log_warn("Could not read the cookie from stdin");
-					break;
+					log_error("Could not read the cookie from stdin\n");
```

Comment:
The extra `\n` seems wrong.

Why `log_warn` → `log_error`? I would keep that for a different PR if needed.

Reply:
`\n` is actually correct, the `log_*` functions do not add it.

Anyway, to me erroring out seems the best way to handle errors here, reason why I changed it from a warning to an error.

Agreed on keeping this out, given that I assume all of this is going to be squashed. I would have kept these changes separate, but a different PR probably makes sense.

Reply:
You're right of course, we can fix all error messages in a different PR.